File: client/src/dataModels/templateLookupItem.ts
import { BaseEntityActiveModel } from "./baseEntityModel";

export class TemplateLookupItem extends BaseEntityActiveModel {
    CompanyId: string;
    JsonValues: string;
    TemplateLookupItemCategoryId: string;
    Name: string;
    Title: string;

    constructor(id: string = "", isActive: boolean = true) {
        super(id, isActive);
        this.CompanyId = null;
        this.JsonValues = "{\"Values\":[]}";
        this.Name = "";
        this.Title = "";
        this.TemplateLookupItemCategoryId = "";
    }

    createFromEntityModel(templateLookupItem: TemplateLookupItem) {
        const newTemplateLookupItem =
            new TemplateLookupItem(templateLookupItem.Id, !!templateLookupItem.IsActive);

        newTemplateLookupItem.CompanyId = templateLookupItem.CompanyId;
        newTemplateLookupItem.Name = templateLookupItem.Name;
        newTemplateLookupItem.Title = templateLookupItem.Title;

        newTemplateLookupItem.JsonValues =
            templateLookupItem.JsonValues;

        newTemplateLookupItem.TemplateLookupItemCategoryId =
            templateLookupItem.TemplateLookupItemCategoryId;

        return newTemplateLookupItem;
    }

}

File: api/Medico.Api/DB/Models/TemplateLookupItem.cs
using System;

namespace Medico.Api.DB.Models
{
    public class TemplateLookupItem : BaseActiveEntity
    {
        public Guid? CompanyId { get; set; }

        public Company Company { get; set; }

        public string Name { get; set; }

        public string JsonValues { get; set; }

        public TemplateLookupItemCategory TemplateLookupItemCategory { get; set; }

        public Guid TemplateLookupItemCategoryId { get; set; }

        public string Title { get; set; }
    }
}
